WiserTaskScheduler/WiserTaskScheduler/Modules/CleanupWiserHistory/Services/CleanupWiserHistoryService.cs

```diff
@@ -76,31 +78,60 @@ public async Task<JObject> Execute(ActionModel action, JObject resultSets, strin
 
-                                                                        DELETE history.*
-                                                                        FROM {tablePrefix}{WiserTableNames.WiserItem} AS item
-                                                                        JOIN {WiserTableNames.WiserHistory} AS history ON history.item_id = item.id AND history.tablename LIKE CONCAT(?tableName, '%') AND history.changed_on < ?cleanupDate
-                                                                        WHERE item.entity_type = ?entityName
-                                                                        """);
+            var dataTable = await databaseConnection.GetAsync($"""
+                                                               SELECT CAST(history.id AS UNSIGNED) AS id
+                                                               FROM {tablePrefix}{WiserTableNames.WiserItem} AS item
+                                                               JOIN {WiserTableNames.WiserHistory} AS history ON history.item_id = item.id AND history.tablename LIKE CONCAT(?tableName, '%') AND history.changed_on < ?cleanupDate
+                                                               WHERE item.entity_type = ?entityName
+                                                               """);
 
-        historyRowsDeleted += await databaseConnection.ExecuteAsync($"""
+            historyEntriesToDelete.AddRange(from DataRow row in dataTable.Rows select row.Field<ulong>("id"));
 
-                                                                     DELETE history.*
-                                                                     FROM {tablePrefix}{WiserTableNames.WiserItem}{WiserTableNames.ArchiveSuffix} AS item
-                                                                     JOIN {WiserTableNames.WiserHistory} AS history ON history.item_id = item.id AND history.tablename LIKE CONCAT(?tableName, '%') AND history.changed_on < ?cleanupDate
-                                                                     WHERE item.entity_type = ?entityName
-                                                                     """);
+            dataTable = await databaseConnection.GetAsync($"""
+                                                           SELECT CAST(history.id AS UNSIGNED) AS id
+                                                           FROM {tablePrefix}{WiserTableNames.WiserItem}{WiserTableNames.ArchiveSuffix} AS item
+                                                           JOIN {WiserTableNames.WiserHistory} AS history ON history.item_id = item.id AND history.tablename LIKE CONCAT(?tableName, '%')AND history.changed_on < ?cleanupDate
+                                                           WHERE item.entity_type = ?entityName
+                                                           """);
+
+            historyEntriesToDelete.AddRange(from DataRow row in dataTable.Rows select row.Field<ulong>("id"));
```

Comment:
Our .NET code conventions call for the usage of the member extension methods instead of the SQL-style query expressions:
> When using LinQ, prefer member extension methods over SQL-style LINQ keywords - e.g. prefer `myList.Where(x)` to `myList where x`.

In this case, the `Select` method could be used.

Reply:
Changed the code.